packages/blockchain-wallet-v4-frontend/src/data/components/debitCard/sagas.ts

```diff
@@ -25,15 +24,40 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
   const filterTerminatedCards = (cards) =>
     cards.filter((card) => card.status !== CardStateType.TERMINATED)
 
+  const getEligibleAccounts = function* (cardId) {
+    try {
+      const data = yield call(api.getDCEligibleAccounts, cardId)
+      yield put(A.setEligibleAccounts(data))
+    } catch (e) {
+      console.error('Failed to get eligible accounts', errorHandler(e))
+    }
+  }
+
+  const getCurrentCardAccount = function* (cardId) {
+    try {
+      yield put(A.getCurrentCardAccountLoading())
+
+      const data = yield call(api.getDCCurrentAccount, cardId)
+      yield put(A.getCurrentCardAccountSuccess(data))
+    } catch (e) {
+      console.error('Failed to get current card account', errorHandler(e))
+      const eligibleAccounts = yield select(selectors.components.debitCard.getEligibleAccounts)
+      yield put(A.getCurrentCardAccountFailure(eligibleAccounts[0]))
```

Comment:
if `elegibleAccounts` for some reason has no length, you'll trigger an error on the error, causing an error

Reply:
Thanks! nice catch, fixed!